README.ksonnet.md

```diff
@@ -0,0 +1,213 @@
+You need a version of ksonnet newer than the [0.7.0 release](https://github.com/ksonnet/ksonnet/releases).
+
+As of this writing, Heptio hasn't released any newer prebuilt binaries so you will need to [build from source](https://ksonnet.io/docs/build-install).
+
+```
+# Clone the ksonnet repo into your GOPATH
+go get github.com/ksonnet/ksonnet
+cd ${GOPATH}/src/github.com/ksonnet/ksonnet
+make install
+```
+
+## Deploy Kubeflow
+
+Initialize a directory to contain your deployment
+```
+ks init my-kubeflow
+```
+
+Install the Kubeflow packages
+
+```
+cd my-kubeflow
+ks registry add kubeflow github.com/google/kubeflow/tree/master/kubeflow
+ks pkg install kubeflow/core
+ks pkg install kubeflow/tf-serving
+ks pkg install kubeflow/tf-job
+```
+
+
+Create the Kubeflow core component. The core component includes 
+  * JupyterHub
+  * TensorFlow job controller
+
+
+```
+ks generate core kubeflow-core --name=kubeflow-core --namespace=${NAMESPACE}
+```
+  * namespace is optional
+  * TODO(jlewi): There's an open github [issue](https://github.com/ksonnet/ksonnet/issues/222) to allow components to pull
+    the namespace from the environment namespace parameter.
+
+
+Define an environment that doesn't use any Cloud features
+  * This environment could be used for minikube or a full K8s cluster that doesn't depend on a cloud features.
+
+```
+ks env add nocloud
+```
+
+The default Kubeflow deployment will be suitable for this no cloud environment so you can just deploy the core components
+
+```
+ks apply nocloud -c kubeflow-core
+```
+
+If the user is running on a Cloud they could create an environment for this.
+
+```
+ks env add cloud
+ks param set --cloud=gke
+```
+   * The cloud parameter triggers a set of curated cloud configs.
+
+They can then deploy to this environment
+
+```
+ks apply cloud -c kubeflow-core
+```
+
+### Serve a model
+
+We treat each deployed model as a [component](https://ksonnet.io/docs/tutorial#2-generate-and-deploy-an-app-component) in your APP.
+
+Create a component for your model
+
+```
+MODEL_COMPONENT=serveInception
+MODEL_NAME=inception
+MODEL_PATH=gs://cloud-ml-dev_jlewi/tmp/inception
+ks generate tf-serving ${MODEL_COMPONENT} --name=${MODEL_NAME} --namespace=default --model_path=${MODEL_PATH}
+```
+
+Deploy it in a particular environment. The deployment will pick up environment parmameters (e.g. cloud) and customize the deployment appropriately
+
+```
+ks apply cloud -c ${MODEL_COMPONENT}
+```
+
+### Submiting a TensorFlow training job
+
+We treat each TensorFlow job as a [component](https://ksonnet.io/docs/tutorial#2-generate-and-deploy-an-app-component) in your APP.
+
+Create a component for your job.
+
+```
+ks generate tf-job ${JOB_NAME} --name=${JOB_NAME}
+```
+
+To configure your job you need to set a bunch of parameters. To see a list of parameters run
+
+```
+ks prototype describe tf-job
+```
+
+Parameters can be set using `ks param` e.g. to set the Docker image used
+
+```
+ks param set ${JOB_NAME} image ${IMAGE}
```

Comment:
Can we instead recommend the declarative method here? I think people would want to version their config as a reproducible artifact, and specifying it this way doesn't allow doing that.

Reply:
It is versioned. ks param set modifies params.jsonnet which is checked into code control.